JunStack.java

```diff
@@ -200,67 +33,30 @@ public void push(String value) {
     public String pop(){
         if (trackStackPosition > 0) {
            String toPop = this.stack[trackStackPosition - 1];
-           System.out.println("Popping String " + toPop + " at index: " + (trackStackPosition - 1));
-           
+
            this.stack[trackStackPosition - 1] = null;
            trackStackPosition -= 1;
            
            return toPop; 
         } else {
-            System.out.println("Stack is empty.");
-            return "Stack is empty.";
+            return null;
         }
     }
     
     public boolean isEmpty(){
-        if (trackStackPosition == 0){
-            return true;
-        } else {
-            return false;
-        }
+       return trackStackPosition == 0;
     }
-    
-    public boolean isFull(){

```

Comment:
The user doesn't care about the internal array, they're using your class so that they don't have to care about arrays and whether or not they're full.